install.sh

```diff
@@ -0,0 +1,877 @@
+    SUDO=sudo
+    if [ $(id -u) -eq 0 ]; then
+        SUDO=
+    fi
+
+    # --- use systemd type if defined or create default ---
+    if [ -n "${INSTALL_RKE2_TYPE}" ]; then
+        SYSTEMD_TYPE=${INSTALL_RKE2_TYPE}
+    else
+        if [ "${CMD_RKE2}" = server ]; then
+            SYSTEMD_TYPE=notify
+        else
+            SYSTEMD_TYPE=exec
+        fi
+    fi
+
+    # --- use binary install directory if defined or create default ---
+    if [ -n "${INSTALL_RKE2_BIN_DIR}" ]; then
+        BIN_DIR=${INSTALL_RKE2_BIN_DIR}
+    else
+        BIN_DIR=/usr/local/bin
+    fi
+
+    # --- use systemd directory if defined or create default ---
+    if [ -n "${INSTALL_RKE2_SYSTEMD_DIR}" ]; then
+        SYSTEMD_DIR="${INSTALL_RKE2_SYSTEMD_DIR}"
+    else
+        SYSTEMD_DIR=/etc/systemd/system
+    fi
+
+    # --- set related files from system name ---
+    SERVICE_RKE2=${SYSTEM_NAME}.service
+    UNINSTALL_RKE2_SH=${UNINSTALL_RKE2_SH:-${BIN_DIR}/${SYSTEM_NAME}-uninstall.sh}
+    KILLALL_RKE2_SH=${KILLALL_RKE2_SH:-${BIN_DIR}/rke2-killall.sh}
+
+    # --- use service or environment location depending on systemd/openrc ---
+    if [ "${HAS_SYSTEMD}" = true ]; then
+        FILE_RKE2_SERVICE=${SYSTEMD_DIR}/${SERVICE_RKE2}
+        FILE_RKE2_ENV=${SYSTEMD_DIR}/${SERVICE_RKE2}.env
+    elif [ "${HAS_OPENRC}" = true ]; then
+        ${SUDO} mkdir -p /etc/rancher/rke2
+        FILE_RKE2_SERVICE=/etc/init.d/${SYSTEM_NAME}
+        FILE_RKE2_ENV=/etc/rancher/rke2/${SYSTEM_NAME}.env
+    fi
+
+    # --- get hash of config & exec for currently installed rke2 ---
+    PRE_INSTALL_HASHES=$(get_installed_hashes)
+
+    # --- if bin directory is read only skip download ---
+    if [ "${INSTALL_RKE2_BIN_DIR_READ_ONLY}" = true ]; then
+        INSTALL_RKE2_SKIP_DOWNLOAD=true
+    fi
+
+    # --- setup channel values
+    INSTALL_RKE2_CHANNEL_URL=${INSTALL_RKE2_CHANNEL_URL:-"https://update.rke2.io/v1-release/channels"}
+    INSTALL_RKE2_CHANNEL=${INSTALL_RKE2_CHANNEL:-"stable"}
+}
+
+# can_skip_download checks if skip download 
+# environment variable set.
+can_skip_download() {
+    if [ "${INSTALL_RKE2_SKIP_DOWNLOAD}" != true ]; then
+        return 1
+    fi
+}
+
+# verify_rke2_is_executable verify an executabe 
+# rke2 binary is installed.
+verify_rke2_is_executable() {
+    if [ ! -x ${BIN_DIR}/rke2 ]; then
+        fatal "executable rke2 binary not found at ${BIN_DIR}/rke2"
+    fi
+}
+
+# setup_verify_arch set arch and suffix, 
+# fatal if architecture not supported.
+setup_verify_arch() {
+    if [ -z "${ARCH}" ]; then
+        ARCH=$(uname -m)
+    fi
+    case ${ARCH} in
+        amd64)
+            ARCH=amd64
+            SUFFIX=
+            ;;
+        x86_64)
+            ARCH=amd64
+            SUFFIX=
+            ;;
+        arm64)
+            ARCH=arm64
+            SUFFIX=-${ARCH}
+            ;;
+        aarch64)
+            ARCH=arm64
+            SUFFIX=-${ARCH}
+            ;;
+        arm*)
+            ARCH=arm
```

Comment:
there is no support for arm yet for rke2

Reply:
Was leaving that as well as that's on the way.